README.md
# Legominds_Bot
My bot for discord using Discord.js

#**All credit goes to: https://discordjs.readthedocs.io/en/latest/**

##You can add this bot to a server using this link:
###- https://discordapp.com/oauth2/authorize?&client_id=212563097565659136&scope=bot
  
Getting Started:
   - Use the default prefix: ## + any command. Example: ##Help
   - Right now all commands are Case Sensitive.


ChangeLog.md
#9/8/16 14:43 -- Legominds
- Created Bot.
- Added basic functionality.
- Created .Bat file.
- Updated Bot's GitHub.
- Registered bot at Discord.
- Tested basic features.
 
#9/8/16 19:20 -- Legominds
- Added Voice, Game and streaming commands.
- Separated code.
- Added more Documentation to GitHub.

#10/8/16 12:10 -- Legominds
- Made the auth ticket more secure (No longer on the main .js file).
- Started making id.js for getting client's voice ID.
- Made it change game to Help command plus how many servers it's on, on startup.
- Disabled "EnableNSFW" until I get around to it.
- Started making the "Prefix" command useable, with error messages.
- Help command tidy up.

#10/8/16 19:24 -- Legominds
- Added a Sleep mode that when active does not allow commands to be sent.
- Fixed up Game and Prefix commands.
 
#11/8/16 21:25 -- Legominds
- Fixed Sleep mode and it is now fully implemented with Prefix + Sleep and Prefix + Awake.
- Added some misc commands, not in Help.
- Starting getting channel id.

#11/8/16 23:25 -- Legominds
- Bot now joins author's voice. Works on multiple guilds.
- Starting on fixing Leave command.
- Added console logging of what voice it's joined + what author initiated the command.
- Started on Help quotes.

#16/8/16 01:58 -- Legominds
- Added more misc/satire commands. ##PogChamp , ##Fgt , ##Feels
- 


LegomindsBot.js
///Variables and background things///
const Discord = require('discord.js');
var client = new Discord.Client();
var prefix = "##"
var bitrate = "96"

client.loginWithToken('', output);

function output(error, token) {
        if (error) {
                console.log('There was an error logging in: ' + error);
                return;
        } else
                console.log('Logged in. Token: ' + token);
}

client.on("ready", function() {
	console.log("Servers: " + client.servers.length);
});

///Non-Voice Commands///

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Ping")){
		client.reply(message, "Pong!");
	}
});


client.on("message", function(message){
	if (message.content.startsWith(prefix + "Help")){
		client.reply(message, "I've sent you some help.");
		client.sendMessage(message.author, "Here you go!");
		client.sendMessage(message.author, "https://github.com/OfficialLegominds/Legominds_Bot/blob/master/Commands.md");
		client.sendMessage(message.author, "Want to add the bot to a server? https://discordapp.com/oauth2/authorize?&client_id=212563097565659136&scope=bot");
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Join")){
		client.reply(message, "Joining Voice.");
		//joinVoiceChannel(channel, callback);
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Leave")){
		client.reply(message, "I have left the Voice.");
		//client.destroy();
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Delete")){
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Servers")){
		client.reply(message, "Servers: " + client.servers.length);
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content === ("Prefix" + "")){
		client.reply(message, "Set the prefix to " + prefix);
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content === ("test")){
		client.reply(message, "'''this is a test'''");
		client.deleteMessage(message);
	}
});

///Voice Stuff///

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Bitrate")){
		client.reply(message, "Current Bitrate: "+ bitrate);
		client.deleteMessage(message);		
	}
	//else{
	//	client.reply(message, "Set the Bitrate to: " + bitrate);
	//}
});


DnD/DnD_Public.js
//Variables and background things//
const Discord = require('discord.js');
var client = new Discord.Client();
var prefix = "@@"
var song = "Nothing"
var nextgame = "Saturday August 20 5:00PM GMT+1"
var Auth = require("./auth.json");
var dndhelp = require("./commands.txt");

//Startup things//

//Auth with Discord//
client.loginWithToken(Auth, output);

function output(error, token) {
        if (error) {
                console.log('There was an error logging in: ' + error);
                return;
        } else
                console.log('Logged in. DnD Bot. Token: ' + token);
}

client.on("ready", function() {
	console.log("Servers: " + client.servers.length);
});


//Makes the bot set game to help//
client.on("ready", function() {
	client.setPlayingGame(prefix + "Help | " +client.servers.length + " Servers");
});


//Non-Voice Commands//

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Ping")){
		client.reply(message, "Pong!");
		console.log("Someone used Ping. " + message.author);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Next")){
		client.reply(message, "The next game is: " + nextgame);
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Help")){
		client.reply(message, "I've sent you some help.");
		client.sendMessage(message.author, dndhelp);
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "GameOn")){
		client.setPlayingGame("DnD on Roll20");
		client.reply(message, "DnD game is now on.");
		client.sendMessage(message, "@everyone The game is now on.");
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "GameOff")){
		client.setPlayingGame(prefix + "Help | " + client.servers.length + " Servers");
		client.reply(message, "The game has ended.");
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Link")){
		client.reply(message, "Here is a link to Roll20");
		client.sendMessage(message.author, "https://roll20.net");
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Servers")){
		client.reply(message, "Servers: " + client.servers.length);
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content === (prefix + "Prefix")){
		client.reply(message, "Please provide a prefix. Current Prefix: " + prefix);
		client.deleteMessage(message);
	}
	if (message.content === (prefix + "Prefix " + "s")){
		client.reply(message, "Set the prefix to " + prefix);
		client.deleteMessage(message);
	}
});


LegomindsBot_Public.js
//Variables and background things//
const Discord = require('discord.js');
var client = new Discord.Client();
var prefix = "##"
var bitrate = "96"
var currentgame = "Not playing anything"
var song = "Nothing"
var streaming = "Discord.js"
var volume = "100"
var Auth = require("./auth.json");
var getid = require("./id.js");
var tempid = ""

//Startup Things//

//Uses Discord Auth to connect to the bot//
client.loginWithToken(Auth, output);

function output(error, token) {
        if (error) {
                console.log('There was an error logging in: ' + error);
                return;
        } else
                console.log('Logged in. Token: ' + token);
}

//Displays how many servers the bot is on//
client.on("ready", function() {
	console.log("Servers: " + client.servers.length);
});

//Makes the bot set game to help//
client.on("ready", function() {
	client.setPlayingGame(prefix + "Help | " +client.servers.length + " Servers");
});



//Non-Voice Commands//

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Ping")){
		client.reply(message, "Pong!" + message.timestamp);
	}
});


client.on("message", function(message){
	if (message.content.startsWith(prefix + "Help")){
		client.reply(message, "I've sent you some help.");
		client.sendMessage(message.author, "Here you go!");
		client.sendMessage(message.author, "https://github.com/OfficialLegominds/Legominds_Bot/blob/master/Commands.md");
		client.sendMessage(message.author, "Want to add the bot to a server? https://discordapp.com/oauth2/authorize?&client_id=212563097565659136&scope=bot");
		client.deleteMessage(message);
	}
});


client.on("message", function(message){
	if (message.content.startsWith(prefix + "Delete")){
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Servers")){
		client.reply(message, "Servers: " + client.servers.length);
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content === (prefix + "Prefix")){
		client.reply(message, "Please provide a prefix. Current Prefix: " + prefix);
		client.deleteMessage(message);
	}
	if (message.content === (prefix + "Prefix " + "s")){
		client.reply(message, "Set the prefix to " + prefix);
		client.deleteMessage(message);
	}
});


//Game and streaming//

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Game")){
		client.setPlayingGame(currentgame);
		client.reply(message, "Set current Game to: " + currentgame);
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Stream")){
		client.setStreaming(streaming, "<Stream Link>", 1)
		client.reply(message, "Set streaming to: " + streaming);
		client.deleteMessage(message);
	}
});


//Voice Stuff//

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Bitrate")){
		client.reply(message, "Current Bitrate: "+ bitrate);
		client.setChannelBitrate("<id here>", bitrate);
		client.deleteMessage(message);		
	}
	//else{
	//	client.reply(message, "Set the Bitrate to: " + bitrate);
	//}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Now")){
		client.reply(message, "Currently playing: " + song);
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Leave")){
		client.reply(message, "I have left Voice.");
		client.leaveVoiceChannel();
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Join")){
		tempid = message.author
		//client.reply(message, "Joining Voice.");
		//client.joinVoiceChannel(message.channel);
		console.log(tempid);
		//client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Play")){
		playFile("C:\Users\ryank\Desktop\song.mp3", volume);
		client.joinVoiceChannel(user);
		client.deleteMessage(message);
	}
});

client.on("message", function(message){
	if (message.content.startsWith(prefix + "Pause")){
		client.reply(message, "Music Paused.");
		pause()
		client.deleteMessage(message);
	}
});


//Misc//

client.on("message", function(message){
	if (message.content.startsWith(prefix + "EnableNSFW")){
		client.reply(message, "You now have access to NSFW areas.");
		client.roles.get("name", "NSFW");
		client.addMemberToRole(message.author, NSFW);
		client.deleteMessage(message);
	}
});

